scripts/stabilization.py

```diff
@@ -0,0 +1,207 @@
+from dolfin import Constant, CompiledExpression, FiniteElement, compile_cpp_code, project, Dx, sqrt
+from classes_EV import entropy, entropy_flux
+
+__all__ = ['StabilizationParameterSD']
+
+_streamline_diffusion_cpp = """
+#include <pybind11/pybind11.h>
+#include <dolfin/function/Expression.h>
+#include <dolfin/function/FunctionSpace.h>
+#include <dolfin/mesh/Vertex.h>
+#include <cmath>
+using namespace dolfin;
+class StabilizationParameterSD : public Expression
+{
+public:
+  // constructor
+  StabilizationParameterSD() : Expression() { }
+
+  // function pointers
+  std::shared_ptr<GenericFunction> J;
+  
+  std::shared_ptr<GenericFunction> J_;
```

Comment:
What is the difference between `J` and `J_`?

Reply:
From line 154, I understand that these are the previous values. Please rename them to e.g. `J_old` and `J_old_old`.